velox/expression/CastExpr.h

```diff
@@ -108,4 +108,38 @@ class CastExpr : public SpecialForm {
   const bool nullOnFailure_;
 };
 
+/// Custom operator for casts from and to custom types.
+class CastOperator {
+ public:
+  virtual ~CastOperator() = default;
+
+  /// Determine whether the cast operator supports casting the custom type to
+  /// the other type or vice versa.
+  virtual bool isSupportedType(const TypePtr& other) const = 0;
+
+  /// Cast an input vector to the custom type.
+  /// @param inputBase The base vector of the decoded input vector
+  /// @param baseRows Non-null rows of inputBase
+  /// @param fromType The type of the original input vector
+  /// @param result The flat output vector of the custom type for inputBase
+  /// (Caller must ensure it is writable.)
+  virtual void castTo(
+      const BaseVector& inputBase,
```

Comment:
inputBase and baseRows can be just input and rows as the implementation doesn't need to know that these are a "base" of something.

Reply:
Makes sense! I removed "base" from parameters in JsonType.cpp throughout. Thanks!